### PR TITLE
feat: added support fro reset_to_default prop

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**Description**: 

- Added Support for `resetToDefault` prop.
- Components are reset to default values using the redux action, `clearValuesToDefault`.

[RS PR](https://github.com/appbaseio/reactivesearch/pull/1725)

[Feature Request PR](https://github.com/appbaseio/reactivesearch/issues/1722)

[Notio Card](https://www.notion.so/appbase/ReactiveSearch-SelectedFilters-reset-to-default-value-99d4a8a722f14947ab41c33a18ee5312)

[Loom Demo: `Latest` Implementation](https://www.loom.com/share/bc69be97442144dbb801251ce999aac3)

[Docs PR](https://github.com/appbaseio/Docs/pull/206)

### CODE REVIEW

File: src/actions/value.js [hunk 1]
```diff
@@ -1,4 +1,13 @@
-import { SET_VALUE, CLEAR_VALUES, PATCH_VALUE, SET_INTERNAL_VALUE } from '../constants';
+import XDate from 'xdate';
```
Comment (bietkul) [inline src/actions/value.js]: @mohdashraf010897 You can not use `xdate` in this package.

File: src/actions/value.js [hunk 2]
```diff
@@ -1,4 +1,13 @@
...
+import { componentTypes } from '../../lib/utils/constants';
```
Comment (bietkul) [inline src/actions/value.js]: Wrong import statements. Use relative paths everywhere.

File: src/actions/value.js [hunk 3]
```diff
@@ -23,6 +32,84 @@ export function setValue(
...
+			} else if (
+				[componentTypes.categorySearch].includes(componentProps[component].componentType)
+			) {
+				valueToSet = componentProps[component].defaultValue
+					? componentProps[component].defaultValue.term
+					: '';
+			} else if (
+				[componentTypes.dateRange].includes(componentProps[component].componentType)
+			) {
+				const formatInputDate = (date) => {
```
Comment (bietkul) [inline src/actions/value.js]: @mohdashraf010897 We should not load the `xdate` here or in `SelectedFilters` component.
You can handle this case in `dateRange` component itself.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: src/actions/value.js (hunk 3 above)
Comment (mohdashraf010897) [reply src/actions/value.js]: 🆗

File: src/actions/value.js (hunk 1 above)
Comment (mohdashraf010897) [reply src/actions/value.js]: 🆗

**Review by bietkul — APPROVED** [review]

**Pull request merged**